roachtest: tpccbench/nodes=9/cpu=4/chaos/partition failed [fix pending]

[(roachtest).tpccbench/nodes=9/cpu=4/chaos/partition failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2693234&tab=buildLog) on [master@83e70ce84b740e27e721c3b73c38a4b8b515094a](https://github.com/cockroachdb/cockroach/commits/83e70ce84b740e27e721c3b73c38a4b8b515094a):

```
		  |
		  | goroutine 7 [chan receive, 598 minutes]:
		  | runtime.gopark(0x515bc60, 0xc00010e118, 0x515170e, 0x2)
		  | 	/usr/local/go/src/runtime/proc.go:306 +0xe5 fp=0xc00009ae90 sp=0xc00009ae70 pc=0x48e245
		  | runtime.chanrecv(0xc00010e0c0, 0xc00009afc0, 0x8412101, 0x515c048)
		  | 	/usr/local/go/src/runtime/chan.go:577 +0x36f fp=0xc00009af20 sp=0xc00009ae90 pc=0x45a3cf
		  | runtime.chanrecv2(0xc00010e0c0, 0xc00009afc0, 0x1)
		  | 	/usr/local/go/src/runtime/chan.go:444 +0x2b fp=0xc00009af50 sp=0xc00009af20 pc=0x45a04b
		  | github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
		  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:98 +0x12c fp=0xc00009afe0 sp=0xc00009af50 pc=0xf740ec
		  | runtime.goexit()
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc00009afe8 sp=0xc00009afe0 pc=0x4c4681
		  | created by github.com/cockroachdb/cockroach/pkg/util/log.init.5
		  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:42 +0x4d
		  |
		  | goroutine 18 [select, 598 minutes, locked to thread]:
		  | runtime.gopark(0x515bec8, 0x0, 0x1809, 0x1)
		  | 	/usr/local/go/src/runtime/proc.go:306 +0xe5 fp=0xc00008ee08 sp=0xc00008ede8 pc=0x48e245
		  | runtime.selectgo(0xc00008ef78, 0xc00008ef70, 0x2, 0x8, 0xc00010e001)
		  | 	/usr/local/go/src/runtime/select.go:338 +0xcef fp=0xc00008ef30 sp=0xc00008ee08 pc=0x49e3af
		  | runtime.ensureSigM.func1()
		  | 	/usr/local/go/src/runtime/signal_unix.go:897 +0x1fa fp=0xc00008efe0 sp=0xc00008ef30 pc=0x4bc9ba
		  | runtime.goexit()
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc00008efe8 sp=0xc00008efe0 pc=0x4c4681
		  | created by runtime.ensureSigM
		  | 	/usr/local/go/src/runtime/signal_unix.go:880 +0xd5
		  |
		  | goroutine 34 [syscall, 598 minutes]:
		  | runtime.notetsleepg(0x8463280, 0xffffffffffffffff, 0x0)
		  | 	/usr/local/go/src/runtime/lock_futex.go:235 +0x34 fp=0xc000088798 sp=0xc000088768 pc=0x460214
		  | os/signal.signal_recv(0x0)
		  | 	/usr/local/go/src/runtime/sigqueue.go:147 +0x9d fp=0xc0000887c0 sp=0xc000088798 pc=0x4c0bdd
		  | os/signal.loop()
		  | 	/usr/local/go/src/os/signal/signal_unix.go:23 +0x25 fp=0xc0000887e0 sp=0xc0000887c0 pc=0xf5ac45
		  | runtime.goexit()
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc0000887e8 sp=0xc0000887e0 pc=0x4c4681
		  | created by os/signal.Notify.func1.1
		  | 	/usr/local/go/src/os/signal/signal.go:150 +0x45
		  |
		  | goroutine 8 [GC worker (idle)]:
		  | runtime.gopark(0x515bd00, 0xc000580000, 0x1418, 0x0)
		  | 	/usr/local/go/src/runtime/proc.go:306 +0xe5 fp=0xc00008f760 sp=0xc00008f740 pc=0x48e245
		  |
		  | stdout:
		Wraps: (8) secondary error attachment
		  | signal: killed
		  | (1) signal: killed
		  | Error types: (1) *exec.ExitError
		Wraps: (9) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *main.withCommandDetails (8) *secondary.withSecondaryError (9) *errors.errorString

```

<details><summary>More</summary><p>


Artifacts: [/tpccbench/nodes=9/cpu=4/chaos/partition](https://teamcity.cockroachdb.com/viewLog.html?buildId=2693234&tab=artifacts#/tpccbench/nodes=9/cpu=4/chaos/partition)
Related:
- #60662 roachtest: tpccbench/nodes=9/cpu=4/chaos/partition failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-60635](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-60635) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #60420 roachtest: tpccbench/nodes=9/cpu=4/chaos/partition failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #60130 roachtest: tpccbench/nodes=9/cpu=4/chaos/partition failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Atpccbench%2Fnodes%3D9%2Fcpu%3D4%2Fchaos%2Fpartition.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

NPE in `relocateOne`, seemingly. cc @tbg 

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1e0c45d]

goroutine 622648 [running]:
panic(0x43f4d40, 0x7a4d0a0)
	/usr/local/go/src/runtime/panic.go:1064 +0x545 fp=0xc01625f7c0 sp=0xc01625f6f8 pc=0x48b205
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0009d2e00, 0x58b9340, 0xc0372da420)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:233 +0x126 fp=0xc01625f820 sp=0xc01625f7c0 pc=0x14d2866
runtime.call32(0x0, 0x518b370, 0xc0270b3c08, 0x1800000018)
	/usr/local/go/src/runtime/asm_amd64.s:540 +0x3e fp=0xc01625f850 sp=0xc01625f820 pc=0x4c2c1e
runtime.reflectcallSave(0xc01625f990, 0x518b370, 0xc0270b3c08, 0x18)
	/usr/local/go/src/runtime/panic.go:881 +0x58 fp=0xc01625f880 sp=0xc01625f850 pc=0x48abf8
runtime.runOpenDeferFrame(0xc0103bf980, 0xc0270b3bc0, 0x0)
	/usr/local/go/src/runtime/panic.go:855 +0x2cd fp=0xc01625f910 sp=0xc01625f880 pc=0x48aaad
panic(0x43f4d40, 0x7a4d0a0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9 fp=0xc01625f9d8 sp=0xc01625f910 pc=0x48ae79
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0009d2e00, 0x58b9340, 0xc0372daf30)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:233 +0x126 fp=0xc01625fa38 sp=0xc01625f9d8 pc=0x14d2866
runtime.call32(0x0, 0x518b370, 0xc0270b3c08, 0x1800000018)
	/usr/local/go/src/runtime/asm_amd64.s:540 +0x3e fp=0xc01625fa68 sp=0xc01625fa38 pc=0x4c2c1e
runtime.reflectcallSave(0xc01625fba8, 0x518b370, 0xc0270b3c08, 0xc000000018)
	/usr/local/go/src/runtime/panic.go:881 +0x58 fp=0xc01625fa98 sp=0xc01625fa68 pc=0x48abf8
runtime.runOpenDeferFrame(0xc0103bf980, 0xc0270b3bc0, 0x0)
	/usr/local/go/src/runtime/panic.go:855 +0x2cd fp=0xc01625fb28 sp=0xc01625fa98 pc=0x48aaad
panic(0x43f4d40, 0x7a4d0a0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9 fp=0xc01625fbf0 sp=0xc01625fb28 pc=0x48ae79
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send.func1(0xc016261a68, 0xc016261b08, 0xc00162ce00, 0xc016261b00)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:100 +0x247 fp=0xc01625fc18 sp=0xc01625fbf0 pc=0x1ebf9e7
runtime.call32(0x0, 0x5186998, 0xc016261398, 0x2000000020)
	/usr/local/go/src/runtime/asm_amd64.s:540 +0x3e fp=0xc01625fc48 sp=0xc01625fc18 pc=0x4c2c1e
panic(0x43f4d40, 0x7a4d0a0)
	/usr/local/go/src/runtime/panic.go:975 +0x47a fp=0xc01625fd10 sp=0xc01625fc48 pc=0x48b13a
runtime.panicmem(...)
	/usr/local/go/src/runtime/panic.go:212
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:742 +0x413 fp=0xc01625fd40 sp=0xc01625fd10 pc=0x4a1f53
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).relocateOne(0xc00162ce00, 0x58b9340, 0xc0372daf90, 0xc02894cbd0, 0xc002e1df60, 0x3, 0x3, 0x84b94d0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_command.go:2929 +0x85d fp=0xc016260368 sp=0xc01625fd40 pc=0x1e0c45d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).relocateReplicas(0xc00162ce00, 0x58b9340, 0xc0372daf90, 0x9d, 0xc0025df090, 0x8, 0x8, 0xc0025df098, 0x8, 0x8, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_command.go:2771 +0x50f fp=0xc016260698 sp=0xc016260368 pc=0x1e0b58f
```

--- a/pkg/kv/kvserver/replica_command.go
+++ b/pkg/kv/kvserver/replica_command.go
@@ -2925,6 +2925,10 @@ func (s *Store) relocateOne(
 			existingNonVoters,
 			s.allocator.scorerOptions(),
 			args.targetType)
+		if targetStore == nil {
+			return nil, nil, fmt.Errorf("none of the remaining %ss %v are legal additions to %v",
+				args.targetType, args.targetsToAdd, desc.Replicas())
+		}
 
 		target := roachpb.ReplicationTarget{
 			NodeID:  targetStore.Node.NodeID,

